# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium/sandbox/win/src/process_mitigations.cc
# Commit: 7ba557fab26e
# Full Hash: 7ba557fab26e27e366372eaf3d78997f71185b20
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-07-25 21:49:06
# Description:
#   Bug 1565848: Revert latest change to MITIGATION_DLL_SEARCH_ORDER. r=aklotz
#   
#   This is until any regressions can be fixed, see bug 1568850.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D39357
# ==============================================================================

diff -r d9ac2124b930 -r 7ba557fab26e security/sandbox/chromium/sandbox/win/src/process_mitigations.cc
--- a/security/sandbox/chromium/sandbox/win/src/process_mitigations.cc	Thu Jul 25 21:26:27 2019 +0300
+++ b/security/sandbox/chromium/sandbox/win/src/process_mitigations.cc	Thu Jul 25 17:44:24 2019 +0000
@@ -77,16 +77,7 @@
 
     // Check for SetDefaultDllDirectories since it requires KB2533623.
     if (set_default_dll_directories) {
-#if defined(COMPONENT_BUILD)
-      const DWORD directory_flags = LOAD_LIBRARY_SEARCH_DEFAULT_DIRS;
-#else
-      // In a non-component build, all DLLs will be loaded manually, or via
-      // manifest definition, so these flags can be stronger. This prevents DLL
-      // planting in the application directory.
-      const DWORD directory_flags =
-          LOAD_LIBRARY_SEARCH_SYSTEM32 | LOAD_LIBRARY_SEARCH_USER_DIRS;
-#endif
-      if (!set_default_dll_directories(directory_flags) &&
+      if (!set_default_dll_directories(LOAD_LIBRARY_SEARCH_DEFAULT_DIRS) &&
           ERROR_ACCESS_DENIED != ::GetLastError()) {
         return false;
       }